tests/integration/helpers/cluster.py

```diff
@@ -4929,6 +4929,19 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.clickhouse_node.start_clickhouse()
 
+# RAII class to pause and unpause container (also to unpause on exit / test failure)
```

Comment:
Maybe, it could be a `contextmanager` method in `Cluster`, but it's already much better than before